logstash-core/src/main/java/org/logstash/ackedqueue/FsUtil.java

```diff
@@ -0,0 +1,43 @@
+package org.logstash.ackedqueue;
+
+import java.io.File;
+import java.io.IOException;
+
+/**
+ * File System Utility Methods.
+ */
+public final class FsUtil {
+
+    private FsUtil() {
+    }
+
+    /**
+     * Checks if the request number of bytes of free disk space are available under the given
+     * path.
+     * @param path Directory to check
+     * @param size Bytes of free space requested
+     * @return True iff the
+     * @throws IOException on failure to determine free space for given path's partition
+     */
+    public static boolean hasFreeSpace(final String path, final long size) throws IOException {
+        final File[] partitions = File.listRoots();
+        File location = new File(path).getCanonicalFile();
+        boolean found = false;
+        while (!found) {
+            if (location == null) {
+                throw new IOException(
+                    String.format("Unable to determine the partition that contains '%s'.", path)
+                );
+            }
+            for (final File partition : partitions) {
+                if (partition.equals(location)) {
+                    found = true;
+                } else {
+                    location = location.getParentFile();
+                }
+            }
+        }
```

Comment:
I don't understand the logic here: 
- the`for` loop, loops through the partitions, if the location is the partition it sets `fount = true` and continue looping, else if it is not, it backtracks and mutates location  **and continue** looping in the remaining partitions.
- upon completing the `for` loop, if `!found` then it repeats the process.

What is confusing is the `location` mutation `location = location.getParentFile();` in the `for` loop. Wouldn't you want to back track the location after the `for` loop only when `!found` ?? something like this:
```java
for (final File partition : partitions) {
    if (partition.equals(location)) { found = true; }
}
if (! found) { location = location.getParentFile(); }
```

Or am I missing something?